zsl/resource/model_resource.py

```diff
@@ -83,11 +88,18 @@ class ResourceQueryContext(object):
     """
 
     def __init__(self, params, args, data):
-        self._params = params
+        # type: (dict, list, dict) -> ()
         self._args = args.copy()
         self._args_original = args
         self._data = data
 
+        # test if params is a list
+        try:
+            len(params)
```

Comment:
As you can see from the SO comments a simple `isinstance` is not enough to test for every variation of "looks like list" in python. There are several magic methods which handles list-like behavior (__len__, __iter__, __getitem__, ..). I did not have time to go trough all of the quirks so I just test for scala/vector variable and add the comment to not confuse readers. Putting it into a function called `is_list(): ...` would be a cleaner solution, but it would only encourage reuse of this quirky code. I would too like a clean simple all around function, but not at that time. What would you suggest?